lib/Elastica/Aggregation/Filter.php

```diff
@@ -13,13 +14,19 @@ class Filter extends AbstractAggregation
 {
     /**
      * @param string         $name
-     * @param AbstractFilter $filter
+     * @param AbstractQuery $filter
      */
-    public function __construct($name, AbstractFilter $filter = null)
+    public function __construct($name, $filter = null)
     {
         parent::__construct($name);
 
         if ($filter !== null) {
+            if ($filter instanceof AbstractFilter) {
```

Comment:
Could we move this to a function in Param for example? This would simplify the code. You could still pass the first part of the message to the function to have it noting what is deprecated.